### PR TITLE
5503 a2ha command mapping

### PR DESCRIPTION
**Pull request by jayvikramsharma1**
### :nut_and_bolt: Description: What code changed, and why?

Mapping of Workspace, secrets, help command in automate from A2HA

Usage:
    chef-automate secrets SUBCOMMAND [ARG] ...
Parameters:
    SUBCOMMAND                       subcommand
    [ARG] ...                        subcommand arguments
Subcommands:
    init                             Generate a new secrets key used to encrypt the secret store.
    set                              Add a new secrets: for more info use 'secrets set --help'
    show                             Show existing secrets: for more info use 'secrets show --help'
    delete                           Delete existing secrets: for more info use 'secrets delete --help'
Args:
	automate_admin_password			Set automate admin password
	sudo_password					Set sudo password to login in machines
	fe_sudo_password				Set different sudo password for frontend machines
	be_sudo_password				Set different sudo password for bacend machines
Options:
    -h, --help                       print help
    -v, --verbose                    Enable verbose output
    -c, --config-file CONFIG_FILE    Load settings from config file (default: "a2ha.rb")

Usage:
    chef-automate [OPTIONS] SUBCOMMAND [ARG] ...
Parameters:
    SUBCOMMAND                       subcommand
    [ARG] ...                        subcommand arguments
Subcommands:
    setup                            Setup a new workspace
    update                           Update existing workspace
    export                           Export workspace bundle for use in other environments
Options:
    -h, --help                       print help
    -v, --verbose                    Enable verbose output
    -c, --config-file CONFIG_FILE    Load settings from config file (default: "a2ha.rb")

### :chains: Related Resources

https://app.zenhub.com/workspaces/the-delta-quadrant-5fc7c4c3922dc10021323fa6/issues/chef/automate/5503

### :+1: Definition of Done

written codes, done developer testing of commands

### :athletic_shoe: How to Build and Test the Change

rebuild components/automate-cli/

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/secrets-ha-template.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    init                             Generate a new secrets key used to encrypt the secret store.
+    set                              Add a new secrets: for more info use 'secrets set --help'
+    show                             Show existing secrets: for more info use 'secrets show --help'
+    delete                           Delete existing secrets: for more info use 'secrets delete --help'
+
+Args:
+	automate_admin_password			Set automate admin password
+	sudo_password					Set sudo password to login in machines
+	fe_sudo_password				Set different sudo password for frontend machines
+	be_sudo_password				Set different sudo password for bacend machines
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/secrets-ha-template.go:21]: spelling mistake -- `should be backend`

File: components/automate-cli/cmd/chef-automate/secrets-ha-template.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Usage:
+    chef-automate secrets SUBCOMMAND [ARG] ...
+
+Parameters:
+    SUBCOMMAND                       subcommand
+    [ARG] ...                        subcommand arguments
+
+Subcommands:
+    init                             Generate a new secrets key used to encrypt the secret store.
+    set                              Add a new secrets: for more info use 'secrets set --help'
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/secrets-ha-template.go:13]: Should be `a new secret`

File: components/automate-cli/cmd/chef-automate/secrets-ha-template.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    show                             Show existing secrets: for more info use 'secrets show --help'
+    delete                           Delete existing secrets: for more info use 'secrets delete --help'
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/secrets-ha-template.go:15]: Can we delete multiple secrets? If no, should be `Delete an existing secret`

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/automate-cli/cmd/chef-automate/secrets-ha.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+func init() {
+	secretsCmd.SetUsageTemplate(secretsHelpDocs)
+	RootCmd.AddCommand(secretsCmd)
+}
+
+var secretsCmd = &cobra.Command{
+	Use:   "secrets",
+	Short: "set secrets to automate HA",
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/secrets-ha.go:21]:
```suggestion
	Short: "Set secrets to automate HA",
```

File: components/automate-cli/cmd/chef-automate/secrets-ha.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	Long:  "set secrets for automate sudo password and admin password in HA mode.",
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/secrets-ha.go:22]:
```suggestion
	Long:  "Set secrets for automate sudo password and admin password in HA mode.",
```

File: components/automate-cli/cmd/chef-automate/workspace-ha-template.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    SUBCOMMAND                       subcommand
+    [ARG] ...                        subcommand arguments
+
+Subcommands:
+    setup                            Setup a new workspace
+    update                           Update existing workspace
+    export                           Export workspace bundle for use in other environments
+
+Options:
+    -h, --help                       print help
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/workspace-ha-template.go:17]:
```suggestion
    -h, --help                       Print help
```

File: components/automate-cli/cmd/chef-automate/workspace-ha-template.go [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package main
+
+var workspaceCommandHelpDocs = `
+Usage:
+    chef-automate [OPTIONS] SUBCOMMAND [ARG] ...
+
+Parameters:
+    SUBCOMMAND                       subcommand
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/workspace-ha-template.go:8]:
```suggestion
    SUBCOMMAND                       Subcommand
```

File: components/automate-cli/cmd/chef-automate/workspace-ha-template.go [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    [ARG] ...                        subcommand arguments
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/workspace-ha-template.go:9]:
```suggestion
    [ARG] ...                        Subcommand arguments
```

File: components/automate-cli/cmd/chef-automate/workspace-ha.go [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+func init() {
+	workspaceCmd.SetUsageTemplate(workspaceCommandHelpDocs)
+	RootCmd.AddCommand(workspaceCmd)
+}
+
+var workspaceCmd = &cobra.Command{
+	Use:   "workspace",
+	Short: "set workspace env for automate HA.",
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/workspace-ha.go:21]:
```suggestion
	Short: "Set workspace env for automate HA.",
```

File: components/automate-cli/cmd/chef-automate/workspace-ha.go [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+	Long:  "set up automate ha cluster workspace.",
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/workspace-ha.go:22]:
```suggestion
	Long:  "Set up automate ha cluster workspace.",
```

File: components/automate-cli/cmd/chef-automate/workspace-ha.go [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+func runWorkspaceCmd(cmd *cobra.Command, args []string) error {
+	return executeWorkspaceCommand(args)
+}
+
+func executeWorkspaceCommand(args []string) error {
+	if len(args) == 0 {
+		writer.Print("please refer \n" + workspaceCommandHelpDocs)
+		return nil
+	}
+	writer.Printf("setting automate HA cluster workspace \n")
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/workspace-ha.go:39]:
```suggestion
	writer.Printf("Setting automate HA cluster workspace \n")
```

**Review by kalroy — CHANGES_REQUESTED** [review]

**Review by vivek-yadav — APPROVED** [review]

**Pull request closed without merging**